### PR TITLE
Handle WMS authentication with encrypted credentials and proxying

### PR DESCRIPTION
**Pull request by mbertrand**
If  a username & password are entered in for a WMS Source, they will be encrypted and saved in the metadata for the source and it's child datasets.  The credentials will then be decrypted, passed on to a proxy, and added to headers of GetMap requests.

A sample encryption key was added to server/constants.py but should be customized for a production instance.

### CODE REVIEW

**Comment by mbertrand** [comment]
@mgrauer @aashish24 ready for review

**Comment by aashish24** [comment]
@mbertrand can you update this branch?

**Comment by mbertrand** [comment]
@aashish24 branch has been updated

File: server/rest/wms_dataset.py [hunk 1]
```diff
@@ -41,14 +43,25 @@ def createWmsDataset(self, wmsSource, params):
         # Legend to be included in the metadata?
```
Comment (mgrauer) [inline server/rest/wms_dataset.py]: would you remove these two comment lines?  these issues have been settled now.

File: web_external/js/views/body/MapPanel.js [hunk 2]
```diff
@@ -11,6 +11,10 @@ minerva.views.MapPanel = minerva.View.extend({
     _specifyWmsDatasetLayer: function (dataset, layer) {
         var minervaMetadata = dataset.getMinervaMetadata();
         var baseUrl = minervaMetadata.base_url;
+        if (minervaMetadata.hasOwnProperty('credentials')) {
+            baseUrl = '/wms_proxy/' + encodeURIComponent(baseUrl) + '/' +
+                    minervaMetadata.credentials;
+        }
         var layerName = minervaMetadata.type_name;
         // TODO: inclued projection in params ??
```
Comment (mgrauer) [inline web_external/js/views/body/MapPanel.js]: Can you remove this comment line also, the projection is included in the params.

**Comment by mgrauer** [comment]
@mbertrand This is fantastic stuff, very nice.

If you can remove those 3 lines worth of comments and rebase onto master then repush, it LGTM.

In the future, can you push a branch directly to this repo?  It'll make it a bit easier to pull down for QA, as well as doing the small fixes like removing the comment lines.

**Comment by mgrauer** [comment]
One other thing I forgot to mention.  I'd like to take the encryption key out of the python and put it in a config file.  There are some existing utilities around config files [here](https://github.com/girder/girder/blob/master/girder/utility/config.py).  

I'm fine with you doing this now or creating an issue for it.

**Comment by mbertrand** [comment]
@mgrauer I removed the comments and rebased to master.  Also created a new issue for moving the encryption key (https://github.com/Kitware/minerva/issues/131).

**Comment by mgrauer** [comment]
Great, LGTM once Travis approves.

**Comment by mgrauer** [comment]
I'm going to merge so I can get #128 rebased and merged.  Sorry to take away your opportunity to push the green button @mbertrand .

:shipit:

**Pull request merged**